src/pages/radio-view/all-tracks-view.js

```diff
@@ -79,6 +88,8 @@ const AllTracksView = () => {
         <>
             <TrackList
                 tracks={tracks}
+                setTracks={setTracks}
+                allTracks={allTracks}
```

Comment:
tracklist only needs the final filteredTracks.
Remove `allTracks={allTracks}` 
`tracks={tracks}` -> `tracks={filteredTracks}`